app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_setting_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(environment, content_path)
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_content_path(environment, content_path)
+      return "#{content_path}/#{::Katello::Glue::Pulp::Repos.repo_path_from_content_path(environment, '')}" if
```

Comment:
In what real case would make sense to generate the url like this?

Reply:
that a user just want to include full path, but thinking over about it, I do not need the `repo_path_from_content_path` method.